### PR TITLE
fix(core/select): reset input text on blur

### LINKED ISSUE #1191: Label is not displaying in Ix Select
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md).
- [X] I have not leaked any internal/restricted information like screenshots, videos, code snippets, links etc.

### What happened?

Hi, I'm using Ix-select. 

Step 1: select any value from dropdown.

step 2: Try to clear the text from the select input field(placeholder).

step 3: Now try to select the previously selected value from drop down. 

 

Expected :: Now I need to see the selected value in placeholder input filed.

Issue :: Previously selected value is not getting displayed after search clearing.
[Ix-select-issue.zip](https://github.com/siemens/ix/files/14783859/Ix-select-issue.zip)

### What type of frontend framework are you seeing the problem on?

Angular

### Which version of iX do you use?

2.0.4

### Code to produce this issue.

```shell
/*
 * SPDX-FileCopyrightText: 2023 Siemens AG
 *
 * SPDX-License-Identifier: MIT
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { Component } from '@angular/core';

@Component({
  selector: 'app-example',
  template: `
    <ix-select name="ix-select-control" [value]="value">
      <ix-select-item label="Item 1" value="1"></ix-select-item>
      <ix-select-item label="Item 2" value="2"></ix-select-item>
      <ix-select-item label="Item 3" value="3"></ix-select-item>
      <ix-select-item label="Item 4" value="4"></ix-select-item>
    </ix-select>
  `,
})
export default class Select {
  value = '1';
}
```

### PR DESCRIPTION
**Pull request by ridvandmrc**
user DOM manipulation to update it

## 💡 What is the current behavior?
 Label is not visible if text is removed from input.

GitHub Issue Number: Closes #1191 

## 🆕 What is the new behavior?

- Now the label is visible if the item selected
-

## 🏁 Checklist

A pull request can only be merged if all of these conditions are met (where applicable):

- [X] 🦮 Accessibility (a11y) features were implemented
- [X] 🗺️ Internationalization (i18n) - no hard coded strings
- [X] 📲 Responsiveness - components handle viewport changes and content overflow gracefully
- [X] 📄 Documentation was reviewed/updated (`pnpm run docs`)
- [X] 🧪 Unit tests were added/updated and pass (`pnpm test`)
- [X] 📸 Visual regression tests were added/updated and pass ([Guide](https://github.com/siemens/ix/blob/main/CONTRIBUTING.md#visual-regression-testing))
- [X] 🧐 Static code analysis passes (`pnpm lint`)
- [X] 🏗️ Successful compilation (`pnpm build`, changes pushed)

### CODE REVIEW

**Comment by danielleroux** [comment]
🙋‍♂️ Please don't ignore the checklist of the pull request after you already checked the points. There is no testcase added.

**Comment by jul-lam** [comment]
Upon review, I identified a minor error. When the select component is utilized with the `editable` property set to `true`, the add field fails to reset.

Animation

**Code:**

```
<ix-select editable value="1">
  <ix-select-item label="Item 1" value="1"></ix-select-item>
  <ix-select-item label="Item 2" value="2"></ix-select-item>
  <ix-select-item label="Item 3" value="3"></ix-select-item>
  <ix-select-item label="Item 4" value="4"></ix-select-item>
</ix-select>
```

**Comment by ridvandmrc** [comment]
> Upon review, I identified a minor error. When the select component is utilized with the `editable` property set to `true`, the add field fails to reset.
> 
> Animation [ Animation ](https://private-user-images.githubusercontent.com/151610373/321590376-80698d5e-8875-40f8-983c-32e1dbe08370.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxODA4MjEsIm5iZiI6MTcxMzE4MDUyMSwicGF0aCI6Ii8xNTE2MTAzNzMvMzIxNTkwMzc2LTgwNjk4ZDVlLTg4NzUtNDBmOC05ODNjLTMyZTFkYmUwODM3MC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNVQxMTI4NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MWY4MjQ3MGIxYmQxODg0OThiN2VkNzhkNmQ2NTk3N2ZlYzc4MjZmNTY0NTE0ZTQ3ZWRjZjVhMGU4YzkwODUzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.r6Vwj64y8eJuy4HfcDyTTskY957E5-ikOu5F5YJc-T4) [ ](https://private-user-images.githubusercontent.com/151610373/321590376-80698d5e-8875-40f8-983c-32e1dbe08370.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTMxODA4MjEsIm5iZiI6MTcxMzE4MDUyMSwicGF0aCI6Ii8xNTE2MTAzNzMvMzIxNTkwMzc2LTgwNjk4ZDVlLTg4NzUtNDBmOC05ODNjLTMyZTFkYmUwODM3MC5naWY_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQxNVQxMTI4NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MWY4MjQ3MGIxYmQxODg0OThiN2VkNzhkNmQ2NTk3N2ZlYzc4MjZmNTY0NTE0ZTQ3ZWRjZjVhMGU4YzkwODUzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.r6Vwj64y8eJuy4HfcDyTTskY957E5-ikOu5F5YJc-T4)
> 
> **Code:**
> 
> ```
> <ix-select editable value="1">
>   <ix-select-item label="Item 1" value="1"></ix-select-item>
>   <ix-select-item label="Item 2" value="2"></ix-select-item>
>   <ix-select-item label="Item 3" value="3"></ix-select-item>
>   <ix-select-item label="Item 4" value="4"></ix-select-item>
> </ix-select>
> ```

Hi,
I don't quite understand the issue, after you click the outside it resets the input, If you select the new item also it is working as expected. can you please share the requirement ?

**Comment by jul-lam** [comment]
Hi @ridvandmrc 

In the given example, the 'add' field (`+ abc`) does not reset. This issue only arises when the ix-select is used in 'editable'. The requirement for 'editable' is to add a new value by clicking the `+ {NEW VALUE}` button, which appears when a novel value is entered. However, the current problem is that this field remains even if the value is reset by clicking outside. For instance, in the example above, the field `+ abc` persists even after resetting to `Item 4`. In this case no 'add' field should be visible, since `Item 4` is not a novel value.

**Comment by ridvandmrc** [comment]
> Hi @ridvandmrc
> 
> In the given example, the 'add' field (`+ abc`) does not reset. This issue only arises when the ix-select is used in 'editable'. The requirement for 'editable' is to add a new value by clicking the `+ {NEW VALUE}` button, which appears when a novel value is entered. However, the current problem is that this field remains even if the value is reset by clicking outside. For instance, in the example above, the field `+ abc` persists even after resetting to `Item 4`. In this case no 'add' field should be visible, since `Item 4` is not a novel value.

Hi,
I see, I have just updated the PR, you can check it again.

**Review by jul-lam — CHANGES_REQUESTED** [review]
Thank you, it looks good! Nonetheless, could you please add one more test for the editable case.

**Comment by ridvandmrc** [comment]
> Thank you, it looks good! Nonetheless, could you please add one more test for the editable case.

Hi @jul-lam @danielleroux ,
New test cases are added related editable select .

File: packages/core/src/components/select/test/select.ct.ts [hunk 1]
```diff
@@ -172,3 +172,107 @@ test('open filtered dropdown on input', async ({ mount, page }) => {
...
+  await expect(add).toBeVisible();
+
+  await page.getByTestId('input').fill('');
+
+  await expect(page.getByRole('button', { name: 'Item 1' })).toBeVisible();
+  await expect(page.getByRole('button', { name: 'Item 2' })).toBeVisible();
+  await expect(add).not.toBeVisible();
+});
+
+test('type an item and click outside then check list again', async ({
```
Comment (jul-lam) [inline packages/core/src/components/select/test/select.ct.ts:244]: Please rename the test to a more descriptive name. For example:

`type in a novel item name in editable mode, click outside and reopen the select`

File: packages/core/src/components/select/test/select.ct.ts [hunk 2]
```diff
@@ -172,3 +172,107 @@ test('open filtered dropdown on input', async ({ mount, page }) => {
...
+  await element.locator('input').fill('Item 2');
+  await expect(item2).toBeVisible();
+
+  await item2.click();
+
+  const inputValue = await element.locator('input').inputValue();
+  expect(inputValue).toEqual('Item 2');
+});
+
+test('type an item name and remove it then check in the list', async ({
```
Comment (jul-lam) [inline packages/core/src/components/select/test/select.ct.ts:213]: Please rename the test to a more descriptive name. For example:

`type in a novel item name in editable mode and then remove it`

File: packages/core/src/components/select/test/select.ct.ts [hunk 3]
```diff
@@ -172,3 +172,107 @@ test('open filtered dropdown on input', async ({ mount, page }) => {
...
+  mount,
+  page,
+}) => {
+  await mount(`
+        <ix-select value="2" editable>
+          <ix-select-item value="1" label="Item 1">Test</ix-select-item>
+          <ix-select-item value="2" label="Item 2">Test</ix-select-item>
+          <ix-select-item value="3" label="Item 3">Test</ix-select-item>
```
Comment (jul-lam) [inline packages/core/src/components/select/test/select.ct.ts:221]: Please remove this line of code, since `Item 3` is not used in your test.

File: packages/core/src/components/select/test/select.ct.ts [hunk 4]
```diff
@@ -172,3 +172,107 @@ test('open filtered dropdown on input', async ({ mount, page }) => {
...
+        <ix-select value="2">
+          <ix-select-item value="1" label="Item 1">Test</ix-select-item>
+          <ix-select-item value="2" label="Item 2">Test</ix-select-item>
+          <ix-select-item value="3" label="Item 3">Test</ix-select-item>
+        </ix-select>
+    `);
+
+  await page.locator('[data-select-dropdown]').click();
+  const element = page.locator('ix-select');
+  await element.evaluate((select: HTMLIxSelectElement) => (select.value = []));
```
Comment (jul-lam) [inline packages/core/src/components/select/test/select.ct.ts:190]: Why do you need this line of code? If it serves no purpose, please remove it. I know I checked this part in my previous review but it seems I missed this one.

**Review by jul-lam — CHANGES_REQUESTED** [review]
Thanks for adding the tests.

File: packages/core/src/components/select/test/select.ct.ts (hunk 4 above)
Comment (ridvandmrc) [reply packages/core/src/components/select/test/select.ct.ts:190]: we can not remove it, the purpose of it that  trying to update select value by programmatically,

Comment (jul-lam) [reply packages/core/src/components/select/test/select.ct.ts:190]: Can't the same thing be achieved by not setting `value="2"` in your html?

Comment (ridvandmrc) [reply packages/core/src/components/select/test/select.ct.ts:190]: I have to do it after rendering the component programatically

Comment (jul-lam) [reply packages/core/src/components/select/test/select.ct.ts:190]: What are you trying to test here? Is this to test whether the dropdown menu stays open after the value is removed programmatically?

Comment (ridvandmrc) [reply packages/core/src/components/select/test/select.ct.ts:190]: the main purpose is changing that updating dropdown value programmatically after first render, @jul-lam

Comment (jul-lam) [reply packages/core/src/components/select/test/select.ct.ts:190]: So the main purpose of the test is:

- render select with a selected value 2
- remove the value programmatically
- set value 2 again (by clicking on "Item 2")
- testing if value 2 is set by checking wether the input text is equal to "Item 2"

If this is the case, please remove 

```
await element.locator('input').fill('Item 3');

const item1 = page.getByRole('button', { name: 'Item 1' });
const item2 = page.getByRole('button', { name: 'Item 2' });
const item3 = page.getByRole('button', { name: 'Item 3' });

await expect(item1).not.toBeVisible();
await expect(item2).not.toBeVisible();
await expect(item3).toBeVisible();
```

since it is not part of the main test, and already testet in the "filter" test. Furthermore, the name of the test should be updated to represent the main test case.

Comment (ridvandmrc) [reply packages/core/src/components/select/test/select.ct.ts:190]: hi @jul-lam , you are right , I have just removed thanks :)

File: packages/core/src/components/select/select.tsx [hunk 5]
```diff
@@ -315,6 +314,8 @@ export class Select {
       this.isDropdownEmpty = this.isEveryDropdownItemHidden;
     } else {
       this.navigationItem = undefined;
+      this.updateSelection();
```
Comment (nuke-ellington) [inline packages/core/src/components/select/select.tsx:317]: Why is this necessary here?

**Review by jul-lam — DISMISSED** [review]

File: packages/core/src/components/select/test/select.ct.ts (hunk 4 above)
Comment (jul-lam) [reply packages/core/src/components/select/test/select.ct.ts:190]: Thank you! PR is approved from my side.

**Review by danielleroux — APPROVED** [review]

**Review by danielleroux — APPROVED** [review]
